### PR TITLE
feat: Typescriptified all the things

### PR DESCRIPTION
**Pull request by kevinwright**
Project now uses end-to-end typescript

### CODE REVIEW

File: .eslintrc [hunk 1]
```diff
@@ -0,0 +1,53 @@
+{
```
Comment (darscan) [inline .eslintrc]: drop eslint

File: package.json [hunk 2]
```diff
@@ -17,16 +20,22 @@
...
   "devDependencies": {
-    "jscs": "^3.0.7",
-    "tap": "^10.3.2",
-    "tap-only": "0.0.5"
+    "@types/handlebars": "^4.0.39",
+    "@types/marked": "^0.4.1",
+    "@types/minimist": "^1.2.0",
+    "@types/node": "^4.0.47",
+    "tap": "github:snyk/node-tap#alternative-runtimes",
+    "tape": "^4.0.0",
```
[the remark this answers is not in the document]
Comment (kevinwright) [reply package.json]: Also gone

File: tsconfig.json [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+        "module": "commonjs",
+        "allowJs": true,
+        "sourceMap": true,
+        "strict": true,
+        "noImplicitAny": false // Needed to compile tap and ansi-escapes
+    },
+    "include": [
+        "./src/**/*",
+    ]
+}
```
Comment (darscan) [inline tsconfig.json]: tiny thing, but you should turn on `"files.insertFinalNewline": true,` in vscode to prevent this

File: tslint.json [hunk 4]
```diff
@@ -0,0 +1,20 @@
+{
```
Comment (darscan) [inline tslint.json]: also, we should probably align this with the tslint config that's been evolving internally

File: tsconfig.json (hunk 3 above)
Comment (kevinwright) [reply tsconfig.json]: Okay, you'll have to show me where to do this though :)

File: tslint.json (hunk 4 above)
Comment (kevinwright) [reply tslint.json]: Agreed - It's a conversation that needs to happen!

**Comment by kevinwright** [comment]
@darscan Regarding the commits... I'll be squashing this at merge time

File: src/index.ts [hunk 5]
```diff
@@ -27,11 +32,11 @@ if (argv.o) { // output destination
   }
 }
 
-snykToHtml.run(source, template, onReportOutput);
+SnykToHtml.run(source, template, onReportOutput);
 
-function onReportOutput(report) {
+function onReportOutput(report: string | void): void {
```
[the remark this answers is not in the document]
Comment (kevinwright) [reply src/index.ts]: Not quite, but this inspired an even better solution

**Pull request merged**